src/utils/with-connection.js

```diff
@@ -0,0 +1,22 @@
+import { getConnection } from 'api/v1/db/oracledb/connection';
+
+/**
+ *
+ * @param {withConnectionClosure} closure must be an async function
+ * @param {object[]} closureArgs arguments to pass to `closure`
+ * @returns {*} the result of calling `closure` with a new `connection` object
+ */
+const withConnection = async (closure, ...closureArgs) => {
+  const connection = await getConnection();
+  return closure(connection, ...closureArgs).finally(() => connection.close());
```

Comment:
Trying to decide if `closureArgs` is premature abstraction or not (it's not currently used by any caller in `pizzas-dao`). Would you prefer this:
```js
const doThing = (connection, query) => await connection.execute(query)

const foo = () => {
  await withConnection(doThing, "SELECT * FROM PIZZAS");
}
```

Or this:
```js
const doThing = (query) => {
  return withConnection((connection) => {
    return await connection.execute(query)
  });
};

const foo = () => {
  await doThing("SELECT * FROM PIZZAS")
}
```

I strongly suspect we'll want to do one of these at some point, because I'm noticing already that there's some duplicate work between `updatePizzaById` and `getPizzaById` in the case where `updatePizzaById` gets a body with no attributes.

Reply:
I think using `closureArgs` should be ok since it should help simplify the code like in your example.

Reply:
Cool, I'll keep it in. Appreciate the sanity check on that.